src/library/vm/vm_json.cpp

```diff
@@ -75,10 +75,13 @@ vm_obj to_obj(json const & j) {
     } else if (j.is_number_float()) {
         float f = j;
         return mk_vm_constructor(json_idx::vfloat, to_obj(f));
-    } else if (j.is_number()) {
-        int i = j;
-        return mk_vm_constructor(json_idx::vint, mk_vm_nat(i));
-    } else if (j.is_string()) {
+    } else if (j.is_number_unsigned()) {
+        std::uint64_t i = j.get<std::uint64_t>();
+        return mk_vm_constructor(json_idx::vint, mk_vm_int(i));
```

Comment:
Do you know which `mk_vm_int` this resolves to?  Is it the `mpz` overload?

Reply:
The middle and top ones are bad and will truncate the value.  Can we just always use `mpz`?

Reply:
It's because `unsigned`/`int` are 32-bit while `uint64_t`/`int64_t` are 64-bit.